### DISCUSSION TITLE
After upgrade v6 to v7: Type 'PluginDefinition' is missing the following properties from type 'BasePlugin<BasePluginEvents, unknown>'

### DISCUSSION THREAD

**Discussion by ApayRus** [katspaugh/wavesurfer.js#3138] [Q&A]
Hi. how to fix this?
```
Type 'PluginDefinition' is missing the following properties from type 'BasePlugin<BasePluginEvents, unknown>': subscriptions, options, onInit, init, and 7 more.ts(2740)
```

**Comment by katspaugh** [comment]
Are you using a custom plugin? In that case you should add those 10 missing methods that the error mentions.

**Reply by ApayRus** [reply]
only Regions and Timeline... 
I tried to update `"@types/wavesurfer.js": "^6.0.6",` up to v7 but it doesn't exist

**Reply by katspaugh** [reply]
You don't need that package for v7, it comes with its own types.

**Reply by ApayRus** [reply]
Regions and Timeline -- are they custom plugins? 
> you should add those 10 missing methods that the error mentions

Can you add more details please. I looked for answer in issues, and haven't found anything.

**Reply by katspaugh** [reply]
By custom I meant not from this repository. So Regions and Timeline aren't custom. Try uninstalling the `@types/wavesurfer.js` package and it should work.

**Reply by ApayRus** [reply]
After uninstall `@types/wavesurfer.js`: 
`Cannot find module 'wavesurfer.js/src/plugin/regions' or its corresponding type declarations.ts(2307)`
![Screenshot from 2023-08-25 11-01-21](https://github.com/katspaugh/wavesurfer.js/assets/1222611/a6d84108-c6d5-4165-8574-b3b5ea3f2f31)

I changed path as on the main page: 
```ts
import Regions from 'wavesurfer.js/plugins/regions'
```
But error is still here.

**Reply by katspaugh** [reply]
What wavesurfer version is it? Please make sure it's 7.1.5.

**Reply by ApayRus** [reply]
yes, it is ... I use it in tsdx project (developing npm module for react). With v6 it worked. Now errors appeared

**Reply by katspaugh** [reply]
If `import Regions from 'wavesurfer.js/plugins/regions'` doesn't work for you, try `import Regions from 'wavesurfer.js/dist/plugins/regions.js'`

**Reply by ApayRus** [reply]
Thank you! 
Now error appears only for `RegionsPluginParams` and `WaveSurferParams`. I don't know from where to import them. 

![Screenshot from 2023-08-25 12-35-19](https://github.com/katspaugh/wavesurfer.js/assets/1222611/a13e310d-2715-49be-8014-1c92257e6733)

**Reply by katspaugh** [reply]
Please see the docs: https://wavesurfer-js.org/docs/modules/plugins_regions

**Reply by ApayRus** [reply]
@katspaugh thank you for fast responses. 
Did I understand right, that instead of: 
`RegionsPluginParams` --> `RegionsPluginOptions`
`WavesurferParams` --> `WavesurferOptions`

For `RegionsPluginOptions` it shows: `type RegionsPluginOptions = undefined`. And that causes errors further. 

![image](https://github.com/katspaugh/wavesurfer.js/assets/1222611/e991676d-8196-40d3-b99f-d43e3262cec3)

**Reply by katspaugh** [reply]
Like I said in the comment above, it should be `'wavesurfer.js/dist/plugins/regions.js'` with `.js` at the end.

**Reply by ApayRus** [reply]
anyway `type RegionsPluginOptions = undefined`

**Reply by ApayRus** [reply]
Even in docs :) 
![Screenshot from 2023-08-25 14-47-28](https://github.com/katspaugh/wavesurfer.js/assets/1222611/a19b62b3-13c7-4cb4-b49b-3c1fec812ab5)

**Reply by katspaugh** [reply]
Ah, true, it's undefined in the code. Because the plugin doesn't take any options.

**Reply by ApayRus** [reply]
Thank you for explanations. I am sorry I didn't notice this line of `readme.md`: 
`The "official" plugins have been completely rewritten and enhanced`. 
it will take time to adjust to the changes.
